### PR TITLE
[TACHYON-1676] Change setAcl to setAttribute at Client side

### PR DESCRIPTION
**Pull request by dongche**
JIRA: https://tachyon.atlassian.net/browse/TACHYON-1676

Change setAcl to setAttribute at Client side and Thrift. 
For server side, make a least change to ensure it works. Since there are also other TODO works about `setState` method, `SetStateEntry`, and master side `SetAttributeOptions`, server side full change could be after these TODO are done, to avoid unnecessary work.

@apc999 , @yupeng9 , @jsimsa , @sundapeng

### CODE REVIEW

**Comment by dongche** [comment]
Jenkins, test this please

File: core/client-internal/src/main/java/tachyon/client/file/options/SetAttributeOptions.java [hunk 1]
```diff
@@ -110,6 +124,66 @@ public boolean getPersisted() {
...
+
+  /**
+   * @return true if the recursive value is set, otherwise false
+   */
+  public boolean hasRecursive() {
+    return mRecursive != null;
+  }
+
+  /**
+   * @return the recursive flag value
```
Comment (jsimsa) [inline core/client-internal/src/main/java/tachyon/client/file/options/SetAttributeOptions.java]: this needs more explanation ... in particular recursive only applies to owner, group, and permission

File: core/client-internal/src/main/java/tachyon/client/file/options/SetAttributeOptions.java [hunk 2]
```diff
@@ -142,6 +216,44 @@ public SetAttributeOptions setPersisted(boolean persisted) {
...
+   */
+  public SetAttributeOptions setPermission(short permission) {
+    mPermission = permission;
+    return this;
+  }
+
+  /**
+   * Sets the recursive flag.
+   *
+   * @param recursive whether to set acl recursively under a directory
```
Comment (jsimsa) [inline core/client-internal/src/main/java/tachyon/client/file/options/SetAttributeOptions.java]: it is not clear what acl refers to here

File: core/server/src/main/java/tachyon/master/file/FileSystemMaster.java [hunk 3]
```diff
@@ -1716,6 +1716,9 @@ public void resetFile(long fileId)
   public void setState(TachyonURI path, SetAttributeOptions options)
       throws FileDoesNotExistException, AccessControlException, InvalidPathException {
     MasterContext.getMasterSource().incSetStateOps(1);
+    if (options.hasOwner() || options.hasGroup() || options.hasPermission()) {
+      setAcl(path, new SetAclOptions(options));
```
Comment (jsimsa) [inline core/server/src/main/java/tachyon/master/file/FileSystemMaster.java]: SetAclOptions? this seems messy ... I have a pending PR that introduces server side SetAttributeOptions (https://github.com/amplab/tachyon/pull/2651) and I suggest we merge that one first and the fix this up

**Comment by jsimsa** [comment]
@dongche thanks for the PR ... left some comments ... in particular, I think this PR should wait until https://github.com/amplab/tachyon/pull/2651 is submitted and then make the appropriate changes on the server-side as well

**Comment by jsimsa** [comment]
as per offline discussion I had with @apc999, I am merging this in now and will address my comments in a separate PR myself

**Pull request merged**